openeo/rest/job.py

```diff
@@ -136,12 +136,40 @@ def get_results(self) -> "JobResults":
         """
         return JobResults(self)
 
-    def logs(self, offset=None) -> List[LogEntry]:
-        """ Retrieve job logs."""
-        # TODO: option to filter on level? Or move filtering functionality to a separate batch job logs class?
+    def logs(
+        self, offset=None, log_level: Optional[Union[int, str]] = None
+    ) -> List[LogEntry]:
+        """Retrieve job logs.
+
+        :param offset: The last identifier (property ``id`` of a LogEntry) the client has received.
+
+            If provided, the back-ends only sends the entries that occurred after the specified identifier.
+            If not provided or empty, start with the first entry.
+
+            Defaults to None.
+
+        :param log_level: Show only messages of this log level and its higher levels.
+
+            You can use either constants from Python's standard module ``logging``
+            or their names (case insensitive).
+
+            For example:
+                ``logging.INFO``, ``"info"`` or ``"INFO"`` can all be used to show the messages
+                for level ``logging.INFO`` and above, i.e. also ``logging.WARNING`` and
+                ``logging.ERROR`` will be included.
+
+            Defaults to ``logging.ERROR`` (also when explicitly passing log_level=None or log_level="").
```

Comment:
Hmm, thinking about this some more, this means that we are changing the original  behavior here and diverging from the behavior implied by the API. 
I think the default behavior of `job.logs()` should be to return all logs (like it originally was) instead of only returning error logs.
Also note that the related jupyter widget also supports on-the-fly filtering, so we should not filter too much up front.


I think we should only limit  the logs printed from a failing batch job, and leave the `.logs()` method untouched.

Reply:
The original behavior, returning all logs, is the default again. 

I also adapted the helper functions `normalize_log_level` and ` string_to_log_level` to return `logging.DEBUG` as a default. I think that also for these helper functions it is better to have a default that will not cause any surprises (when new code eventually starts to use those functions), and `logging.DEBUG` would be the value that works like a no-operation. Seems safer.